binderhub/repoproviders.py

```diff
@@ -294,6 +295,44 @@ def get_build_slug(self):
         return "figshare-{}".format(self.record_id)
 
 
+class DataverseProvider(RepoProvider):
```

Comment:
This all looks pretty reasonable to me. One thing I'd ask is if you could write out some short documentation for the spec for this provider. Could you:

1. Add more contextual information to the docstrings in this class
2. Add an entry for the Dataverse URL structure to the BinderHub docs here: https://binderhub.readthedocs.io/en/latest/api.html#providers ?
3. (optionally) not necessary for this PR but, add one for the other repoproviders that are support but not listed there, if you want extra kudos :-)

Reply:
I believe that's fixed by Phil's commit.